walkeeper/src/safekeeper.rs

```diff
@@ -154,24 +154,24 @@ pub struct VoteResponse {
 /// announces 1) successful election (with epoch_start_lsn); 2) commit_lsn.
 #[derive(Debug, Serialize, Deserialize)]
 pub struct AppendRequest {
-    h: AppendRequestHeader,
-    wal_data: Bytes,
+    pub h: AppendRequestHeader,
+    pub wal_data: Bytes,
 }
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct AppendRequestHeader {
-    term: Term,
+    pub term: Term,
     // LSN since the proposer appends WAL; determines epoch switch point.
-    epoch_start_lsn: Lsn,
+    pub epoch_start_lsn: Lsn,
     /// start position of message in WAL
-    begin_lsn: Lsn,
+    pub begin_lsn: Lsn,
     /// end position of message in WAL
-    end_lsn: Lsn,
+    pub end_lsn: Lsn,
     /// LSN committed by quorum of safekeepers
-    commit_lsn: Lsn,
+    pub commit_lsn: Lsn,
     /// minimal LSN which may be needed by proposer to perform recovery of some safekeeper
-    truncate_lsn: Lsn,
+    pub truncate_lsn: Lsn,
     // only for logging/debugging
-    proposer_uuid: PgUuid,
+    pub proposer_uuid: PgUuid,
```

Comment:
> `pub`

Not sure how important here, but feels quite odd to make a few of those fields publicly available.
Might make changing the code later a bit harder when this all leaks being used as a public api.

I notice that you use those only to create a struct, so maybe instead of making every field public, we can have another `pub create_for_test(..) -> Self` method?

Reply:
It also feels quite odd to me. These messages are part of walproposer <-> safekeeper protocol, and some of them are already declared public. I think there may be some sense to make all these structures public, because they are already used in walproposer as identical C definitions, but not really sure.